src/producer.rs

```diff
@@ -15,6 +16,10 @@ pub enum ExtractCodeError {
     Producer(#[from] ProducerError),
     #[error(transparent)]
     Schema(#[from] SchemaError),
+    #[error(transparent)]
+    VectorRequestError(#[from] reqwest::Error),
+    #[error("Vector request failed with status: {0}")]
+    VectorRequestStatusError(reqwest::StatusCode),
```

Comment:
I think we could name this differently, since how this is used right now is jsut for when the channel is closed